salt/modules/zypper.py

```diff
@@ -1195,6 +1203,7 @@ def install(name=None,
         __zypper__(no_repo_failure=ignore_repo_failure).call(*cmd)
 
     __context__.pop('pkg.list_pkgs', None)
+    __context__.pop('pkg.list_provides', None)
```

Comment:
I still don't like this **very much** and I would still rather move away this at least into a `def _remove_methods()` function. If someone got it strange, we should not amplify it, but refactor instead:
```python
def _remove_methods():
    '''
    Removes methods that do not needs to be in __context__ for some reasons.
    '''
    # This part we can extend then only here for everywhere
    for method in ['list_pkgs', 'list_provides']:
        __context__.pop('pkg.{0}'.format(method), None)

# And then somewhere later:
...
__zypper__(systemd_scope.........
_remove_methods()
```